docs/components/contacts.rst

```diff
@@ -1,4 +1,257 @@
 Contacts
 ==========================================================
 
+There's several ways to extend Contacts in Mautic.
+One of them is to show custom events in a Contact's event timeline - this document shows you how.
 
+Contact timeline/history
+------------------------
+
+To inject events into a Contact's timeline, create an event listener that listens to the ``LeadEvents::TIMELINE_ON_GENERATE`` event.
+Using this event, the plugin can inject unique items into the timeline and also into the engagements graph on each page.
+
+.. note:: Before using this event listener, you'll need to ensure that you store your custom events in a custom database table. See :ref:`Generating timeline events from your own custom events` below for more details.
+
+The event listener receives a ``Mautic\LeadBundle\Event\LeadTimelineEvent`` object. You can find the commonly used methods below the code example.
+
+.. code-block:: PHP
+
+    <?php
+    // plugins/HelloWorldBundle/EventListener/LeadSubscriber.php
+
+    declare(strict_types=1);
+
+    namespace MauticPlugin\HelloWorldBundle\EventListener;
+
+    use Doctrine\ORM\EntityManager;
+    use Mautic\LeadBundle\Event\LeadTimelineEvent;
+    use Mautic\LeadBundle\LeadEvents;
+    use MauticPlugin\HelloWorldBundle\Entity\WorldRepository;
+    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+    use Symfony\Component\Routing\RouterInterface;
+    use Symfony\Contracts\Translation\TranslatorInterface;
+
+    class LeadSubscriber implements EventSubscriberInterface
```

Comment:
Migrated from `CommonSubscriber` (which no longer exists) to EventSubscriberInterface, including proper Dependency Injection